### PR TITLE
Use `readr::read_csv()` throughout the lesson

### PR DESCRIPTION
**Pull request by Teebusch**
Currently, "starting with data" introduces  base R's `read.csv()`, but "Manipulating data" then switches to `readr::read_csv()`.
SInce R 4.0 the default behavior of both functions is to read strings as character vectors, so there is little reason to introduce both. For beginners the confusion that this creates is arguably larger than the insights gained from it, as evidenced by some of the issues here.

This is related to a number of issues and might close some of them:
#609 #471 #608 #553 #555 #559 #616 #645 

The main change is:

- move to "starting with data" section: explanation of packages and package loading, introduction of tibbles, 
- remove discussion of "stringsAsFactors" (except for a short mention of former default behavior)
- adapt challenges and code examples
- smaller rewriting here and there.

### CODE REVIEW

File: 02-starting-with-data.Rmd [hunk 1]
```diff
@@ -50,66 +50,102 @@ Each row holds information for a single animal, and the columns represent:
...
-Inside the download.file command, the first entry is a character string with the source URL 
-("https://ndownloader.figshare.com/files/2292169"). This source URL downloads a CSV file from 
-figshare. The text after the comma ("data_raw/portal_data_joined.csv") is the destination of the 
-file on your local machine. You'll need to have a folder on your machine called "data_raw" where 
-you'll download the file. So this command downloads a file from Figshare, names it 
-"portal_data_joined.csv" and adds it to a preexisting folder named "data_raw".
+that contains the survey data from Figshare, and we will use `read_csv()` to
+load the content of the CSV file into R.
+
+Inside the download.file command, the first entry is a character string with the
```
Comment (fmichonneau) [inline 02-starting-with-data.Rmd:59]:
```suggestion
Inside the `download.file` command, the first entry is a character string with the
```

**Review by fmichonneau — COMMENTED** [review]
Thank you so much for taking the time to work on this @Teebusch!
My only suggestion would be to remove (or make an aside) the comparison between the base `data.frame` object and the `tibble` given that most of the workshop participants will not know what a `data.frame` is and the comparison will add cognitive load.
Everything else is a great improvement to the current content. So thanks again!

**Comment by fmichonneau** [comment]
would another @datacarpentry/r-ecology-lesson-maintainers be willing to also review this PR? Thanks!

File: 02-starting-with-data.Rmd [hunk 2]
```diff
@@ -134,15 +171,24 @@ factors). For example, here is a figure depicting a data frame comprising a
...
+differences are that:
+
+1. In addition to displaying the data type of each column under its name, it
+   only prints the first few rows of data and only as many columns as fit on one
+   screen. If you want to see more rows, you can use the `print()` function 
+2. Columns of class `character` are never converted into factors. This is 
+   different from the default behavior of R in versions prior to 4.0.
+   
+
 ## Inspecting `data.frame` Objects
```
[the remark this answers is not in the document]
Comment (Teebusch) [reply 02-starting-with-data.Rmd]: fixed in 339684c211f8f936404d43ca2488777279658616

File: 02-starting-with-data.Rmd [hunk 3]
```diff
@@ -134,15 +171,24 @@ factors). For example, here is a figure depicting a data frame comprising a
...
-
-We can see this when inspecting the <b>str</b>ucture of a data frame
+We can see this also when inspecting the <b>str</b>ucture of a data frame
 with the function `str()`:
 
 ```{r, purl=FALSE}
 str(surveys)
 ```
 
+You might have noticed that our data frame is actually a "tibble". A tibble is a
```
[the remark this answers is not in the document]
Comment (Teebusch) [reply 02-starting-with-data.Rmd]: I agree! See my suggested changes in 4991253e9513aaaa9433441f143cc569e54ec96e

**Comment by mondorescue** [comment]
> 
> 
> would another @datacarpentry/r-ecology-lesson-maintainers be willing to also review this PR? Thanks!

@fmichonneau I'll review this PR as a second.

**Review by mondorescue — APPROVED** [review]
All the changes looks good to me.

**Comment by fmichonneau** [comment]
Thank you both!

**Pull request merged**